scipy/stats/_multivariate.py

```diff
@@ -3850,3 +3852,452 @@ def rvs(self, dim, size=1, random_state=None):
+    scipy.stats.multinomial : The multinomial distribution.
+
+    References
+    ----------
+    .. [1] The Multivariate Hypergeometric Distribution,
+           http://www.randomservices.org/random/urn/MultiHypergeometric.html
+
+    .. [2] Thomas J. Sargent and John Stachurski, 2020,
+           Multivariate Hypergeometric Distribution
+           https://python.quantecon.org/_downloads/pdf/multi_hyper.pdf
+    """  # noqa: E501
+    def __init__(self, seed=None):
+        super(multivariate_hypergeom_gen, self).__init__(seed)
+        self.__doc__ = doccer.docformat(self.__doc__, mhg_docdict_params)
+
+    def __call__(self, m, n, seed=None):
+        """
+        Create a frozen multivariate_hypergeom distribution.
+
+        See `multivariate_hypergeom_frozen` for more information.
+        """
+        return multivariate_hypergeom_frozen(m, n, seed=seed)
+
+    def _process_parameters(self, m, n):
+        m = np.asarray(m, dtype=np.int_)
+        n = np.asarray(n, dtype=np.int_)
+        if m.ndim == 0:
+            raise ValueError("'m' must be an array!")
+
+        m, n = np.broadcast_arrays(m, n[..., np.newaxis])
+        n = n[..., 0]
+        mcond = m < 0
+
+        M = m.sum(axis=-1)
+
+        ncond = (n <= 0) | (n > M)
+        return M, m, n, mcond, ncond, np.any(mcond, axis=-1) | ncond
+
+    def _process_quantiles(self, x, M, m, n):
+        xx = np.asarray(x, dtype=np.int_)
+        if xx.ndim == 0:
+            raise ValueError("x must be an array.")
+        if xx.size != 0 and not xx.shape[-1] == m.shape[-1]:
+            raise ValueError(f"Size of each quantile should be size of 'm': "
+                             f"received {xx.shape[-1]}, "
+                             f"but expected {m.shape[-1]}.")
+
+        xx, m, n, M = np.broadcast_arrays(xx, m,
+                                          n[..., np.newaxis],
+                                          M[..., np.newaxis])
+        n, M = n[..., 0], M[..., 0]
+
+        xcond = xx != x
+        xcond = xcond | (xx < 0) | (xx > m)
+        return (xx, M, m, n, xcond,
+                np.any(xcond, axis=-1) | (xx.sum(axis=-1) != n))
+
+    def _checkresult(self, result, cond, bad_value):
+        result = np.asarray(result)
+        if cond.ndim != 0:
+            result[cond] = bad_value
+        elif cond:
+            return bad_value
+        if result.ndim == 0:
+            return result[()]
+        return result
+
+    def _logpmf(self, x, M, m, n, mxcond, ncond):
+        # This equation of the pmf comes from the relation,
+        # n combine r = beta(n+1, 1) / beta(r+1, n-r+1)
+        num = np.zeros_like(m, dtype=np.float_)
+        den = np.zeros_like(n, dtype=np.float_)
+        num[~mxcond] = (betaln(m[~mxcond]+1, 1) -
+                        betaln(x[~mxcond]+1, m[~mxcond]-x[~mxcond]+1))
+        den[~ncond] = (betaln(M[~ncond]+1, 1) - betaln(n[~ncond]+1,
+                       M[~ncond]-n[~ncond]+1))
```

Comment:
Not too big a deal, but logical indexing creates a copy. Better to create the copies once than twice, and it makes it more readable like it was originally.
```suggestion
        m, x, M, n = m[~mxcond], x[~mxcond], M[~ncond], n[~ncond]
        num[~mxcond] = betaln(m +1, 1) - betaln(x + 1, m - x + 1)
        den[~ncond] = betaln(M + 1, 1) - betaln(n + 1, M - n + 1)
```

Reply:
Fixed.